modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -778,6 +796,32 @@ public void deleteRelatedObjectEntries(
 
 	@Override
 	public ObjectEntry expireObjectEntry(
+			long userId, long objectEntryId, ServiceContext serviceContext)
+		throws PortalException {
+
+		ObjectEntry objectEntry = objectEntryPersistence.findByPrimaryKey(
+			objectEntryId);
+
+		objectEntry = updateStatus(
+			userId, objectEntry, WorkflowConstants.STATUS_EXPIRED,
+			serviceContext);
+
+		List<ObjectEntryVersion> objectEntryVersions =
+			_objectEntryVersionLocalService.getObjectEntryVersions(
+				objectEntry.getObjectEntryId());
+
+		if (!objectEntryVersions.isEmpty()) {
+			for (ObjectEntryVersion objectEntryVersion : objectEntryVersions) {
+				_objectEntryVersionLocalService.expireObjectEntryVersion(
+					userId, objectEntryVersion);
+			}
+		}
+
+		return objectEntry;
+	}
+
+	@Override
+	public ObjectEntry expireObjectEntryVersion(
```

Comment:
@JhossephG delete this method and call the expireObjectEntryVersion from _objectEntryVersionLocalService when needed